pkg/controllers/common/constants.go

```diff
@@ -194,12 +211,15 @@ var (
 	DeploymentGVR = appsv1.SchemeGroupVersion.WithResource(DeploymentResource)
 	DaemonSetGVR  = appsv1.SchemeGroupVersion.WithResource(DaemonSetResource)
 	ReplicaSetGVR = appsv1.SchemeGroupVersion.WithResource(ReplicaSetResource)
+
+	ServiceExportGVR = mcsv1alpha1.SchemeGroupVersion.WithResource(ServiceExportResource)
+	EndpointSliceGVR = discoveryv1b1.SchemeGroupVersion.WithResource(EndpointSliceResource)
 )
 
-// MaxFederatedObjectNameLength defines the max length of a federated object name.
+// MaxNameLength defines the max length of a kubernetes resource name.
```

Comment:
This length only applies to CRDs.

Reply:
I tested some of Kubernetes native resources. Seems their max name length is also 253. I'm not sure which resources are not this length.

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names